RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -0,0 +1,210 @@
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+
+#include "Math/GenVector/VectorUtil.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include <map>
+#include <vector>
+
+template< typename T>
+class L1TJetsMatching: public edm::global::EDProducer<> {
+ public:
+  explicit L1TJetsMatching(const edm::ParameterSet&);
+  ~L1TJetsMatching();
+  virtual void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+ private:
+    
+  const edm::EDGetTokenT<std::vector<T>> jetSrc_;
+  const edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs> jetTrigger_;
+  const double pt1Min;
+  const double pt2Min;
+  const double mjjMin;
+  const double matchingR;
+  const double matchingR2;
+  };
+    //
+    // class decleration
+    //
+    using namespace reco   ;
+    using namespace std    ;
+    using namespace edm    ;
+    using namespace trigger;
+    
+    
+    template< typename T>
+    std::pair<std::vector<T>,std::vector<T>> categorise(const std::vector<T>& pfMatchedJets, double pt1, double pt2, double Mjj)
+    {
+        std::pair<std::vector<T>,std::vector<T>> output;
+        unsigned int i1 = 0;
+        unsigned int i2 = 0;
+        double mjj = 0;
+        if (pfMatchedJets.size()>1){
+            for (unsigned int i = 0; i < pfMatchedJets.size()-1; i++){
+                
+                const T &  myJet1 = (pfMatchedJets)[i];
+                
+                for (unsigned int j = i+1; j < pfMatchedJets.size(); j++)
+                {
+                    const T &  myJet2 = (pfMatchedJets)[j];
+                    
+                    const double mjj_test = (myJet1.p4()+myJet2.p4()).M();
+                    
+                    if (mjj_test > mjj){
+                        
+                        mjj =mjj_test;
+                        i1 = i;
+                        i2 = j;
+                    }
+                }
+            }
+            
+            const T &  myJet1 = (pfMatchedJets)[i1];
+            const T &  myJet2 = (pfMatchedJets)[i2];
+            
+            if ((mjj > Mjj) && (myJet1.pt() >= pt1) && (myJet2.pt() > pt2) )
+            {
+                
+                output.first.push_back(myJet1);
+                output.first.push_back(myJet2);
+                
+            }
+            
+            if ((mjj > Mjj) && (myJet1.pt() < pt1) && (myJet1.pt() > pt2) && (myJet2.pt() > pt2))
+            {
```

Comment:
This is the three jets case, where the two jets forming the max mjj have to have pt more than 35GeV but the higher pt one (myJets1) has pt<95 (otherwise it would have been the above two jets case).